src/coreclr/nativeaot/Runtime/inc/CommonTypes.h

```diff
@@ -6,10 +6,34 @@
 
 #include <cstddef>
 #include <cstdint>
-#include <cstdlib>
-#include <new>
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifdef TARGET_WINDOWS
 
+#include <new>
 using std::nothrow;
+
+#else
+
+extern "C" void __cxa_pure_virtual();
+
+namespace nativeaot
+{
+    struct nothrow_t {};
+    const nothrow_t nothrow;
+}
+
+void* operator new(size_t n, const nativeaot::nothrow_t&) noexcept;
+void* operator new[](size_t n, const nativeaot::nothrow_t&) noexcept;
+void operator delete(void *p) noexcept;
+void operator delete[](void *p) noexcept;
```

Comment:
Is it possible to make the operator delete to have a local scope somehow?

Reply:
I haven't spent much time but the initial attempt was failing when I put it in the namespace. (this is how we do it in `util_nodependencies.cpp`). Will try again.

Reply:
Unfortunately operators are not allowed in namespaces. It is possible to declare them in classes, but we can't use them with `using` declaration; we get `error: using declaration cannot refer to class member`.